### PR TITLE
add "TIL How to Select Merge with Ecto.Query" post

### LINKED ISSUE #1953: TIL blog post: How to use Ecto.Query#select_merge/3
Hello! I'd love to cross-post a recent post I wrote for the flatiron labs blog about my new fave Ecto query function, `Ecto.Query#select_merge/3`: https://medium.com/flatiron-labs/til-how-to-select-merge-with-ecto-query-679d03204b9d

It includes examples of how and when to use it, as well as pros/cons versus another (totally valid) approach like adding more schema associations.

### PR DESCRIPTION
**Pull request by ktravers**
closes #1953

### CODE REVIEW

File: _posts/2019-08-23-til-ecto-select-merge.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+To start with, we have an `AdmissionEvent` schema that looks like this:
+
+```elixir
+defmodule Registrar.Tracking.AdmissionEvent do
+  use Ecto.Schema
+
+  schema "admission_events" do
+    field(:action, :string)
+    field(:admission_id, :integer)
+    field(:admitter_uuid, :uuid)
```
Comment (tajchumber) [inline _posts/2019-08-23-til-ecto-select-merge.md]: I think you use `uuid` in the migration file and `Ecto.UUID` when defining the schema
so the line should be `field(:admitter_uuid, Ecto.UUID)`

File: _posts/2019-08-23-til-ecto-select-merge.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+```
+
+...and a User schema that looks like this:
+
+```elixir
+defmodule Registrar.User do
+  use Ecto.Schema
+
+  schema "users" do
+    field(:uuid, :uuid)
```
Comment (tajchumber) [inline _posts/2019-08-23-til-ecto-select-merge.md]: same here as above

File: _posts/2019-08-23-til-ecto-select-merge.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+  def for_admission(query \\ AdmissionEvent, admission) do
+    from(ae in query,
+      where: ae.admission_id == ^admission.id,
+      order_by: [desc: ae.occurred_at],
+
+      #### STEP TWO ####
+      #  Join on User  #
+      ##################
+      join: u in User,
+      on: r.admitter_uuid == u.uuid,
```
Comment (tajchumber) [inline _posts/2019-08-23-til-ecto-select-merge.md]: where does the `r` variable come from in the query? I couldn't get the code to compile.  I got this error:
```
** (Ecto.Query.CompileError) unbound variable `r` in query. If you are attempting to interpolate a value, use ^var
    lib/ecto/query.ex:755: Ecto.Query.from/5
    expanding macro: Ecto.Query.from/2
    /tracking/admission_event.ex:22: EctoSelectMerge.Tracking.AdmissionEvent.for_admission/2
    (elixir) lib/kernel/parallel_compiler.ex:229: anonymous fn/4 in Kernel.ParallelCompiler.spawn_workers/7
```

Could you please clarify this? :)

File: _posts/2019-08-23-til-ecto-select-merge.md (hunk 1 above)
Comment (ktravers) [reply _posts/2019-08-23-til-ecto-select-merge.md]: i would have thought that too, based on the documentation about [Ecto "custom types"](https://hexdocs.pm/ecto/Ecto.Schema.html#module-custom-types). but my team has an app running in production that defines uuid fields same as above, and we haven't run into any issues. i'm wondering if maybe it has something to do with `Ecto.UUID.type()` being `:uuid`.

that said, if you think folks will find it confusing, i'm happy to change it to `Ecto.UUID`

File: _posts/2019-08-23-til-ecto-select-merge.md (hunk 3 above)
Comment (ktravers) [reply _posts/2019-08-23-til-ecto-select-merge.md]: yep, will fix. it's a copy pasta mistake :)

File: _posts/2019-08-23-til-ecto-select-merge.md (hunk 1 above)
Comment (tajchumber) [reply _posts/2019-08-23-til-ecto-select-merge.md]: The code didn't compile for me when the field type is `:uuid`
The error I got was: 
```
** (ArgumentError) invalid or unknown type :uuid for field :admitter_uuid
```

File: _posts/2019-08-23-til-ecto-select-merge.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+      order_by: [desc: ae.occurred_at]
+    )
+  end
+end
+```
+
+```elixir
+# Taking our query function for a spin...
+
+iex> admission = Repo.get(Admission, 1)
```
Comment (tajchumber) [inline _posts/2019-08-23-til-ecto-select-merge.md]: Didn't notice this before, but there isn't a definition for `Admission`
I think the tutorial would be simple if had one 

What do you think?

**Comment by doomspork** [comment]
Hey @ktravers this is awesome! Sorry I've been out of commission recently. I'll shoot out a tweet about your post this week 🎉

**Pull request merged**